scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/DocumentOps.scala

```diff
@@ -316,24 +316,44 @@ trait DocumentOps { self: DatabaseOps =>
                 }
               }
 
+              object ForComprehensionImplicitArg {
+                private def isForComprehensionSyntheticName(select: g.Select): Boolean = {
+                  select.pos == select.qualifier.pos && (select.name == g.nme.map ||
```

Comment:
We should also include `foreach` and add a test for it with a `for (...) y` (no yield).